src/dataflow/components/dataflow-program.tsx

```diff
@@ -302,14 +325,18 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
 
     const programData = {
       program: {
-        endTime: newTimestamp,
+        endTime: programEndTime,
         hubs,
         program: editedProgram,
         programId: programName,
         runInterval: interval * 1000,
         sensors
       }
     };
+
+    this.props.onSetProgramRunId(programName);
+    this.props.onSetProgramStartEndTime(programStartTime, programEndTime);
+
```

Comment:
This is temporary. Will need to make a new document and set these values. However, to facilitate showing data and setting UI state, we will set values on the current tile in the open document for now.